src/base/bittorrent/magneturi.cpp

```diff
@@ -31,29 +31,18 @@
 #include <libtorrent/bencode.hpp>
 #include <libtorrent/error_code.hpp>
 #include <libtorrent/magnet_uri.hpp>
+#include <libtorrent/sha1_hash.hpp>
 
-#include <QByteArray>
 #include <QRegularExpression>
-#include <QStringList>
-
-#include "base/utils/string.h"
 
 namespace
 {
-    QString bcLinkToMagnet(QString bcLink)
+    bool isBitTorrentInfoHash(const QString &string)
     {
-        QByteArray rawBc = bcLink.toUtf8();
-        rawBc = rawBc.mid(8); // skip bc://bt/
-        rawBc = QByteArray::fromBase64(rawBc); // Decode base64
-        // Format is now AA/url_encoded_filename/size_bytes/info_hash/ZZ
-        QStringList parts = QString(rawBc).split('/');
-        if (parts.size() != 5) return QString();
-
-        QString filename = parts.at(1);
-        QString hash = parts.at(3);
-        QString magnet = "magnet:?xt=urn:btih:" + hash;
-        magnet += "&dn=" + filename;
-        return magnet;
+        return (((string.size() == (libtorrent::sha1_hash::size * 2))
```

Comment:
Why not hardcode `40` since those are the characters of a sha1 digest(?) when represented as string?
It is more intuitive IMO in this situation.

Reply:
It was `40` before, changed because I think exactly otherwise...

Reply:
>A user at first glance won't understand why look at the size and multiply it by 2. 

Maybe that is a good thing, makes one thinks more... 

>He will have to delve into libtorrent code, see the actual size and come to the realization we need a 40-char string here because "sha1".

The advantage is the reader doesn't need to bother about any numbers or hash length except only understanding 1 byte expands to 2 hex chars.

>Seeing 40 he will think "uhm we're looking at a info hash which is sha1 digest aka 40 chars".

This is the opposite of me, I took seconds to realize why the number `40`.

>Maybe add code comments?

Yeah I added some, please take a look.

Reply:
Which one do you prefer? Now there is 1 vote on each side:
1. `40`
2. `(libtorrent::sha1_hash::size * 2)`

>different approach in these branches. 

This is even more exotic for base32 case: `(libtorrent::sha1_hash::size * 1.6)`, where `1.6` is the efficiency for converting bytes to base32.

Reply:
>This is even more exotic for base32 case

Since SHA1 size will never changed I prefer constants in this code (of course with comments you added above it).

Reply:
>I prefer constants in this code 

You mean `40` & `32`?

Reply:
Can we document by storing these values in well-named local variables?

Reply:
Yes.
Or, maybe, one time calculated named constants (e.g. `constexpr int SHA1_BASE32_SIZE = libtorrent::sha1_hash::size * 1.6; constexpr int SHA1_HEX_SIZE = libtorrent::sha1_hash::size * 2;` at top of file).

Reply:
>Can we document by storing these values in well-named local variables?
>
>Or, maybe, one time calculated named constants 

OK and OK, just one thing `constexpr` doesn't imply `const` in c++14 and compiler will very very likely to evaluate it at build-time even when just using `const`.
PR updated.